### PR TITLE
GLT-4260 Display Requisition warnings

### PR DESCRIPTION
**Pull request by Christopher-Ho**
Jira ticket or GitHub issue: https://jira.oicr.on.ca/browse/GLT-4260

- [ X ] Includes a change file

### CODE REVIEW

File: miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/Dtos.java [hunk 1]
```diff
@@ -489,6 +490,15 @@ public static SampleDto asDto(@Nonnull Sample from, boolean includeBoxPositions,
       setId(dto::setRequisitionId, requisition);
       setString(dto::setRequisitionAlias, requisition.getAlias());
       dto.setRequisitionAssayIds(requisition.getAssays().stream().map(Assay::getId).toList());
+      dto.setRequisitionStopped(requisition.isStopped());
+      for (RequisitionPause pause: requisition.getPauses())
+      {
+        LocalDate endDate = pause.getEndDate();
+        if (endDate == null || endDate.isAfter(LocalDate.now()))
+        {
+          dto.setRequisitionPaused(true);
```
Comment (avarsava) [inline miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/Dtos.java:499]: you can `break` out of the loop once you find the first reason to set this to true - the rest is a waste of cycles

Comment (avarsava) [reply miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/Dtos.java:499]: Alternatively, stream the pauses and use anyMatch()

File: miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/Dtos.java [hunk 2]
```diff
@@ -614,6 +624,15 @@ private static DetailedSampleDto asDetailedSampleDto(@Nonnull DetailedSample fro
       setId(dto::setEffectiveRequisitionId, requisition);
       setString(dto::setEffectiveRequisitionAlias, requisition.getAlias());
       dto.setRequisitionAssayIds(requisition.getAssays().stream().map(Assay::getId).toList());
+      dto.setRequisitionStopped(requisition.isStopped());
+      for (RequisitionPause pause: requisition.getPauses())
+      {
+        LocalDate endDate = pause.getEndDate();
+        if (endDate == null || endDate.isAfter(LocalDate.now()))
+        {
+          dto.setRequisitionPaused(true);
```
Comment (avarsava) [inline miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/Dtos.java:633]: ditto for all following instances

File: miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/Dtos.java (hunk 1 above)
Comment (djcooke) [reply miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/Dtos.java:499]: This is also repeated in 5 places. Extract to a private static method

**Review by avarsava — APPROVED** [review]

File: miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/Dtos.java [hunk 3]
```diff
@@ -1493,13 +1501,21 @@ public static LibraryDto asDto(@Nonnull Library from, boolean includeBoxPosition
     if (requisition != null) {
       setId(dto::setRequisitionId, requisition);
       setString(dto::setRequisitionAlias, requisition.getAlias());
+      dto.setRequisitionStopped(requisition.isStopped());
+      dto.setRequisitionPaused(requisition.getPauses().stream().anyMatch(
+          pause -> (pause.getEndDate() == null || pause.getEndDate().isAfter(LocalDate.now()))
+      ));
```
Comment (djcooke) [inline miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/Dtos.java:1507]: Remove these lines. Already being done below outside of the if/else that handles direct vs. effective requisitions

**Review by djcooke — APPROVED** [review]

**Pull request merged**